code/modules/clothing/head/misc_special.dm

```diff
@@ -218,3 +218,22 @@
 	hair_style = pick(GLOB.hair_styles_list - "Bald") //Don't want invisible wig
 	hair_color = "#[random_short_color()]"
 	. = ..()
+
+/obj/item/clothing/head/foilhat
+	name = "tinfoil hat"
+	desc = "Thought control rays, psychotronic scanning. Don't mind that, I'm protected cause I made this hat."
+	icon_state = "foilhat"
+	item_state = "foilhat"
+	armor = list("melee" = 0, "bullet" = 0, "laser" = -5,"energy" = 0, "bomb" = 0, "bio" = 0, "rad" = -5, "fire" = 0, "acid" = 0)
+
+/obj/item/clothing/head/foilhat/equipped(mob/living/carbon/human/user)

```

Comment:
This triggers whenever you equip the hat in any slot, use the `slot` argument to see if it's equipped on the head yet
https://github.com/tgstation/tgstation/blob/8f26a27df70ffd13df88d90f8cba7e40161b6fdf/code/game/objects/items.dm#L431